test/e2e/framework/bind.go

```diff
@@ -54,25 +57,33 @@ func Bind(t *testing.T, authURLCh chan<- string, invocations chan<- SubCommandIn
 	require.NoError(t, err)
 	err = opts.Validate()
 	require.NoError(t, err)
-	opts.Runner = func(cmd *exec.Cmd) error {
-		bs, err := io.ReadAll(cmd.Stdin)
-		if err != nil {
-			return err
-		}
-		if invocations != nil {
-			invocations <- SubCommandInvocation{
-				Executable: cmd.Args[0],
-				Args:       cmd.Args[1:],
-				Stdin:      bs,
+
+	if !opts.DryRun {
+		opts.Runner = func(cmd *exec.Cmd) error {
+			bs, err := io.ReadAll(cmd.Stdin)
+			if err != nil {
+				return err
 			}
-		}
-		t.Logf("Running command: %s\nstdin:\n", cmd.String())
-		t.Logf("%s", bs)
+			if invocations != nil {
+				invocations <- SubCommandInvocation{
+					Executable: cmd.Args[0],
+					Args:       cmd.Args[1:],
+					Stdin:      bs,
+				}
+			}
+			t.Logf("Running command: %s\nstdin:\n", cmd.String())
+			t.Logf("%s", bs)
 
-		return nil
+			return nil
+		}
 	}
 	err = opts.Run(ctx, authURLCh)
 	require.NoError(t, err)
+
+	if opts.DryRun {
+		_, err = yaml.YAMLToJSON(bufOut.Bytes())
```

Comment:
Currently this only verifies one resource yaml. I'm going to add multiple binding resource support into e2e suite but it requires a little bit effort.